### PR TITLE
Update HipChat plugin to support self-hosted servers

### PR DESCRIPTION
**Pull request by bertramtruong**
This PR will allow users to specify a "target" for hipchat_notifier so that they can send notifications to a self-hosted [HipChat Server](https://www.hipchat.com/server) instance.

### CODE REVIEW

**Comment by mavimo** [comment]
Please add test to check the correct behavior when the target is set.

Also add a check connection error if hipchat config is wrong.

**Comment by dancryer** [comment]
Hey @bertramtruong - Please could you rebase this change on top of the current master branch? 

Otherwise, :+1: from me.

File: PHPCI/Plugin/HipchatNotify.php [hunk 1]
```diff
@@ -52,6 +53,12 @@ public function __construct(Builder $phpci, Build $build, array $options = array
                 $this->message = Lang::get('x_built_at_x');
             }
 
+            if (isset($options['target'])) {
```
Comment (mavimo) [inline PHPCI/Plugin/HipchatNotify.php]: This should be done by ternary operator:
```php
 $this->target = isset($options['target']) ? $options['target'] : "https://api.hipchat.com";
```

**Comment by mavimo** [comment]
@bertramtruong any chance to be update / rebased?

**Pull request closed without merging**